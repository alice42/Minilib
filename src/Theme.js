export const theme = {
  backgroundColor: '#ECEFF1',
  onBackground: '#000000',
  surfaceColor: '#FFFFFF',
  surfaceColorHover: '#F5F5F5',
  surfaceColorActive: '#E0E0E0',
  onSurface: '#000000',
  primaryColor: '#0277BD',
  primaryColorHover: '#0288D1',
  primaryColorActive: '#01579B',
  onPrimary: '#FFFFFF',
  secondaryColor: '#00897B',
  secondaryColorHover: '#00695C',
  secondaryColorActive: '#004D40',
  onSecondary: '#FFFFFF',
  tertiaryColor: '#EF6C00',
  tertiaryColorHover: '#FB8C00',
  tertiaryColorActive: '#E65100',
  onTertiary: '#FFFFFF',
  fontPrimary: {
    regular: 'Lato,sans-serif',
    bold: 'Lato,sans-serif',
    light: 'Lato,sans-serif'
  },
  fontSecondary: {
    regular: 'arial,sans-serif',
    bold: 'arial,sans-serif',
    light: 'arial,sans-serif'
  },
  fontTertiary: {
    regular: 'Georgia,serif',
    bold: 'Georgia,serif',
    light: 'Georgia,serif'
  },
  font48: '4rem',
  font43: '3.6rem',
  font38: '3.2rem',
  font34: '2.9rem',
  font30: '2.5rem',
  font29: '2.4rem',
  font28: '2.3rem',
  font26: '2.1rem',
  font24: '2rem',
  font23: '1.9rem',
  font22: '1.85rem',
  font20: '1.65rem',
  font19: '1.6rem',
  font18: '1.5rem',
  font17: '1.4rem',
  font16: '1.35rem',
  font15: '1.25rem',
  font14: '1.2rem',
  font13: '1.1rem',
  font12: '1rem',
  font11: '0.95rem',
  font10: '0.9rem',
  font9: '0.75rem',
  font8: '0.7rem',
  font6: '0.5rem',
  overlayOpacity: 0.7,
  borderRadius: '5px',
  border: 'solid 5px #000000'
}
