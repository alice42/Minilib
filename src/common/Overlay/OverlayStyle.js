import styled, { css } from 'styled-components'
import { transparentize } from 'polished'

const OverlayStyle = styled.div`
  display: flex;
  flex-direction: column;
  position: absolute;
  align-self: center;
  ${({ rounded, theme }) => rounded && css`
    border-radius: ${typeof rounded === 'string' ? rounded : theme.borderRadius};
  `}
  padding: ${({ padding }) => padding};
  /* color switching with dark light ou color props */
  color: ${({ dark, light, color, backgroundColor, theme }) => dark ? (backgroundColor || theme.backgroundColor) : light ? (color || theme.onBackground) : color || theme.onSurface};
  /* Background color default transparent. Using transparentize (1 - opacity) to apply to any type of color (ex: rgb(), rgba(), #000000, #000, blue, ...) */
  background-color: ${({ dark, light, backgroundColor, color, opacity, theme }) => transparentize(
    1.0 - (opacity !== undefined ? parseFloat(opacity) : parseFloat(theme.overlayOpacity)),
    dark ? (color || theme.onBackground) : light ? (backgroundColor || theme.backgroundColor) : backgroundColor || 'transparent'
  )};
  /* justify and align flex if fullsize is on */
  justify-content: ${({ fullsize, top, bottom }) => fullsize ? (top && !bottom ? 'flex-start' : (bottom && !top ? 'flex-end' : 'center')) : 'center'};
  align-items: ${({ fullsize, left, right }) => fullsize ? (left && !right ? 'flex-start' : (right && !left ? 'flex-end' : 'center')) : 'center'};

  top: ${({ top, fullsize, margin }) => fullsize ? margin : top ? (typeof top === 'boolean' ? margin : top) : 'auto'};
  bottom: ${({ bottom, fullsize, margin }) => fullsize ? margin : bottom ? (typeof bottom === 'boolean' ? margin : bottom) : 'auto'};
  left: ${({ left, fullsize, margin }) => fullsize ? margin : left ? (typeof left === 'boolean' ? margin : left) : 'auto'};
  right: ${({ right, fullsize, margin }) => fullsize ? margin : right ? (typeof right === 'boolean' ? margin : right) : 'auto'};

  /* if fullsize, we need to handle top, bottom, etc... props differently to configure padding instead of margin */ 
  ${({ fullsize, top, bottom, left, right }) => fullsize && css`
    ${top && (typeof top === 'string') && css`padding-top: ${top};`}
    ${bottom && (typeof bottom === 'string') && css`padding-bottom: ${bottom};`}
    ${left && (typeof left === 'string') && css`padding-left: ${left};`}
    ${right && (typeof right === 'string') && css`padding-right: ${right};`}
  `}
  ${({ customStyle }) => css(customStyle)}
`

export default OverlayStyle
