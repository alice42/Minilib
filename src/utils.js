export let conflictsValidator = (validType, conflicts) => (props, propName, componentName) => {
  let conflictedProp
  if (typeof props[propName] !== validType) return new Error(`<${componentName}>: ${propName} must be a ${validType}!`)
  if (props[propName] && (conflictedProp = conflicts.find(p => props[p]))) {
    return new Error(`<${componentName}>: ${propName} can't be used if ${conflictedProp} is already used!`)
  }
}
