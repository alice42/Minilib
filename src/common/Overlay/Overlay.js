import React, { Component } from 'react'
import PropTypes from 'prop-types'
import OverlayStyle from './OverlayStyle'
import { conflictsValidator } from '../../utils'

class Overlay extends Component {
  static propTypes = {
    /** set dark background and light text color */
    dark: conflictsValidator('boolean', ['light']),
    /** set light background and dark text color */
    light: conflictsValidator('boolean', ['dark']),
    /** round borders style on overlay (ex: true, "10px") */
    rounded: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** text color */
    color: PropTypes.string,
    /** background color (need an opacity (:dark, :light or :opacity)) */
    backgroundColor: PropTypes.string,
    /** opacity of the overlay (ex: "0.8") */
    opacity: PropTypes.number,
    /** overlay takes all the space it can, or takes the minimum it needs */
    fullsize: PropTypes.bool,
    /** overlay displayed on top, mixable with others positions (ex: true, "20px") */
    /** modifying margin if fullsize disabled and padding when enabled */
    top: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** overlay displayed on bottom, see :top */
    bottom: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** overlay displayed on left, see :top */
    left: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** overlay displayed on right, see :top */
    right: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** padding inside the overlay (ex: "15px") */
    padding: PropTypes.string,
    /** margin around, directions (top, left, ...) takes the priority on it */
    margin: PropTypes.string,
    /** custom style object */
    customStyle: PropTypes.object
  }
  static defaultProps = {
    dark: false,
    light: false,
    rounded: false,
    fullsize: true,
    top: false,
    bottom: false,
    left: false,
    right: false,
    padding: '10px',
    margin: '0px',
    customStyle: {}
  }

  render () {
    return (
      <OverlayStyle {...this.props}>
        {this.props.children}
      </OverlayStyle>
    )
  }
}

export default Overlay
