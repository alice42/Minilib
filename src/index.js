import Button from './common/Buttons/Button'
import Icon from './common/Icons/Icon'
import Image from './common/Image/Image'
import InputText from './common/Inputs/InputText/InputText'
import ExternalLink from './common/Links/ExternalLink'
import Overlay from './common/Overlay/Overlay'
import Video from './common/Video/Video'

export { Button }
export { Icon }
export { Image }
export { InputText }
export { ExternalLink }
export { Overlay }
export { Video }
