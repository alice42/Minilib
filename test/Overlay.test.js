import React from 'react'
import Overlay from '../src/common/Overlay/Overlay'
import 'jest-styled-components'
import { renderCreateWithThemeJSON } from './TestUtils'
import { theme } from '../src/Theme'
import { transparentize } from 'polished'

test('display', () => {
  const comps = [
    renderCreateWithThemeJSON(<Overlay />),
    renderCreateWithThemeJSON(<Overlay><p>Hello</p><span>World</span><h2>!!!</h2></Overlay>)
  ]
  expect(comps[0]).not.toBeNull()
  expect(comps[0].children).toBeNull()
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).not.toBeNull()
  expect(comps[1].children.length).toEqual(3)
  expect(comps[1]).toMatchSnapshot()
})

test(':top :left :right :bottom :fullsize', () => {
  const comps = [
    renderCreateWithThemeJSON(<Overlay fullsize />),
    renderCreateWithThemeJSON(<Overlay fullsize={false} top left right='23px' />),
    renderCreateWithThemeJSON(<Overlay fullsize right bottom='23px' />),
    renderCreateWithThemeJSON(<Overlay fullsize top='10px' left='0px' right='10%' />),
    renderCreateWithThemeJSON(<Overlay fullsize left />),
    renderCreateWithThemeJSON(<Overlay fullsize={false} bottom />),
    renderCreateWithThemeJSON(<Overlay fullsize={false} top='10px' bottom='15px' right left='5em' />)
  ]
  expect(comps[0]).toHaveStyleRule('justify-content', 'center')
  expect(comps[0]).toHaveStyleRule('align-items', 'center')
  expect(comps[0]).toHaveStyleRule('top', '0px')
  expect(comps[0]).toHaveStyleRule('bottom', '0px')
  expect(comps[0]).toHaveStyleRule('left', '0px')
  expect(comps[0]).toHaveStyleRule('right', '0px')
  expect(comps[0]).not.toHaveStyleRule('padding-top')
  expect(comps[0]).toMatchSnapshot()

  expect(comps[1]).toHaveStyleRule('justify-content', 'center')
  expect(comps[1]).toHaveStyleRule('align-items', 'center')
  expect(comps[1]).toHaveStyleRule('top', '0px')
  expect(comps[1]).toHaveStyleRule('bottom', 'auto')
  expect(comps[1]).toHaveStyleRule('left', '0px')
  expect(comps[1]).toHaveStyleRule('right', '23px')
  expect(comps[1]).not.toHaveStyleRule('padding-bottom')
  expect(comps[1]).toMatchSnapshot()

  expect(comps[2]).toHaveStyleRule('justify-content', 'flex-end')
  expect(comps[2]).toHaveStyleRule('align-items', 'flex-end')
  expect(comps[2]).toHaveStyleRule('top', '0px')
  expect(comps[2]).toHaveStyleRule('bottom', '0px')
  expect(comps[2]).toHaveStyleRule('left', '0px')
  expect(comps[2]).toHaveStyleRule('right', '0px')
  expect(comps[2]).not.toHaveStyleRule('padding-top')
  expect(comps[2]).toHaveStyleRule('padding-bottom', '23px')
  expect(comps[2]).toMatchSnapshot()

  expect(comps[3]).toHaveStyleRule('justify-content', 'flex-start')
  expect(comps[3]).toHaveStyleRule('padding-top', '10px')
  expect(comps[3]).toHaveStyleRule('padding-left', '0px')
  expect(comps[3]).toHaveStyleRule('padding-right', '10%')
  expect(comps[3]).toMatchSnapshot()
  expect(comps[4]).toHaveStyleRule('align-items', 'flex-start')

  expect(comps[5]).toHaveStyleRule('bottom', '0px')
  expect(comps[5]).toHaveStyleRule('top', 'auto')
  expect(comps[5]).toHaveStyleRule('left', 'auto')
  expect(comps[5]).toHaveStyleRule('right', 'auto')
  expect(comps[5]).toMatchSnapshot()

  expect(comps[6]).toHaveStyleRule('bottom', '15px')
  expect(comps[6]).toHaveStyleRule('top', '10px')
  expect(comps[6]).toHaveStyleRule('left', '5em')
  expect(comps[6]).toHaveStyleRule('right', '0px')
  expect(comps[6]).toMatchSnapshot()
})

test(':rounded', () => {
  const comps = [
    renderCreateWithThemeJSON(<Overlay />),
    renderCreateWithThemeJSON(<Overlay rounded />),
    renderCreateWithThemeJSON(<Overlay rounded={'87px'} />)
  ]
  expect(comps[0]).not.toHaveStyleRule('border-radius')
  expect(comps[1]).toHaveStyleRule('border-radius', theme.borderRadius)
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toHaveStyleRule('border-radius', '87px')
})

test(':colors', () => {
  const comps = [
    renderCreateWithThemeJSON(<Overlay />),
    renderCreateWithThemeJSON(<Overlay dark />),
    renderCreateWithThemeJSON(<Overlay dark backgroundColor={'blue'} color={'white'} />),
    renderCreateWithThemeJSON(<Overlay light />),
    renderCreateWithThemeJSON(<Overlay light opacity={0.1} backgroundColor={'blue'} color={'white'} />),
    renderCreateWithThemeJSON(<Overlay backgroundColor={'blue'} color={'white'} />)
  ]
  expect(comps[0]).toHaveStyleRule('background-color', 'transparent')
  expect(comps[0]).toHaveStyleRule('color', theme.onSurface)
  expect(comps[0]).toMatchSnapshot()

  expect(comps[1]).toHaveStyleRule('background-color', transparentize(1.0 - theme.overlayOpacity, theme.onBackground))
  expect(comps[1]).toHaveStyleRule('color', theme.backgroundColor)
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toHaveStyleRule('background-color', transparentize(1.0 - theme.overlayOpacity, 'white'))
  expect(comps[2]).toHaveStyleRule('color', 'blue')

  expect(comps[3]).toHaveStyleRule('background-color', transparentize(1.0 - theme.overlayOpacity, theme.backgroundColor))
  expect(comps[3]).toHaveStyleRule('color', theme.onBackground)
  expect(comps[3]).toMatchSnapshot()
  expect(comps[4]).toHaveStyleRule('background-color', transparentize(1.0 - 0.1, 'blue'))
  expect(comps[4]).toHaveStyleRule('color', 'white')
  expect(comps[4]).toMatchSnapshot()
  expect(comps[5]).toHaveStyleRule('background-color', transparentize(1.0 - theme.overlayOpacity, 'blue'))
  expect(comps[5]).toHaveStyleRule('color', 'white')
})
