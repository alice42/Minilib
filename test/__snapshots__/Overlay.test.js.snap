// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:colors 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
/>
`;

exports[`:colors 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #ECEFF1;
  background-color: rgba(0,0,0,0.7);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
/>
`;

exports[`:colors 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: rgba(236,239,241,0.7);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
/>
`;

exports[`:colors 4`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: white;
  background-color: rgba(0,0,255,0.1);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
  color="white"
  opacity={0.1}
/>
`;

exports[`:rounded 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border-radius: 5px;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
/>
`;

exports[`:top :left :right :bottom :fullsize 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
/>
`;

exports[`:top :left :right :bottom :fullsize 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: auto;
  left: 0px;
  right: 23px;
}

<div
  className="c0"
/>
`;

exports[`:top :left :right :bottom :fullsize 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  padding-bottom: 23px;
}

<div
  className="c0"
/>
`;

exports[`:top :left :right :bottom :fullsize 4`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  padding-top: 10px;
  padding-left: 0px;
  padding-right: 10%;
}

<div
  className="c0"
/>
`;

exports[`:top :left :right :bottom :fullsize 5`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: auto;
  bottom: 0px;
  left: auto;
  right: auto;
}

<div
  className="c0"
/>
`;

exports[`:top :left :right :bottom :fullsize 6`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 10px;
  bottom: 15px;
  left: 5em;
  right: 0px;
}

<div
  className="c0"
/>
`;

exports[`display 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
/>
`;

exports[`display 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

<div
  className="c0"
>
  <p>
    Hello
  </p>
  <span>
    World
  </span>
  <h2>
    !!!
  </h2>
</div>
`;
